File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    // Time complexity O(n)
    // Space complexity O(1)
    void reverseString(vector<char>& s) {
        int start = 0;
        int end = s.size() - 1;

        while(start < end) {
          char tmp = s[start];
          s[start++] = s[end];
          s[end--] = tmp;
        }
    }
};

File: 0344-reverse-string/0344-reverse-string.java
class Solution {
    // time O(n) + space O(1)
    public void reverseString(char[] s) {
        int start = 0;
        int end = s.length - 1;
        while(start < end) {
            char tmp = s[start];
            s[start] = s[end];
            s[end] = tmp;
            start++;
            end--;
        }
    }
}